File: app/api/schemas/text.py
"""
Text Processing API Schemas
"""

from pydantic import BaseModel
from typing import Optional


class TextProcessRequest(BaseModel):
    """Text processing request"""
    text: str
    emotion: Optional[str] = "neutral"
    session_id: Optional[str] = None
    return_audio: bool = False


class TextProcessResponse(BaseModel):
    """Text processing response"""
    response_text: str
    audio_url: Optional[str] = None
    processing_time: float

File: frontend/src/api/text.ts
import apiClient from './client';
import { TextProcessResponse, EmotionType } from '../types';

export async function processText(
  text: string,
  emotion: EmotionType = 'neutral',
  sessionId?: string,
  returnAudio: boolean = false
): Promise<TextProcessResponse> {
  const response = await apiClient.post('/text/process', {
    text,
    emotion,
    session_id: sessionId,
    return_audio: returnAudio,
  });

  return response.data;
}

export async function processTextWithSession(
  sessionId: string,
  text: string,
  emotion: EmotionType = 'neutral',
  returnAudio: boolean = false
): Promise<TextProcessResponse> {
  const response = await apiClient.post(`/text/process/${sessionId}`, {
    text,
    emotion,
    return_audio: returnAudio,
  });

  return response.data;
}
